File: docs/screenshots/take-screenshot.py
#!/usr/bin/env python3
"""Take a settings tab screenshot of the Daily Reflect desktop app."""

import os, pathlib
from playwright.sync_api import sync_playwright

ROOT = pathlib.Path(__file__).resolve().parent.parent.parent
HTML_PATH = ROOT / "apps" / "agent" / "src" / "index.html"
OUT_DIR = ROOT / "docs" / "screenshots"

TAURI_SHIM = """
window.__TAURI__ = {
  core: { invoke: async () => ({}) },
  event: { listen: async () => () => {} },
  shell: { open: async () => {} },
};
"""

SHOW_SETTINGS = """
// Hide onboarding
const onb = document.getElementById("onboarding-screen");
if (onb) onb.classList.add("hidden");

// Show tab bar
const tabBar = document.querySelector(".tab-bar");
if (tabBar) tabBar.style.display = "";

// Activate settings tab
document.querySelectorAll(".tab-content").forEach(c => c.classList.remove("active"));
document.getElementById("tab-settings").classList.add("active");

// Show settings main panel
document.querySelectorAll(".view-panel").forEach(p => p.classList.remove("active"));
document.getElementById("settings-main").classList.add("active");

// Tab bar icon
document.querySelectorAll(".tab-bar-item").forEach(b => b.classList.remove("active"));
document.querySelector('.tab-bar-item[data-nav="settings"]').classList.add("active");

// Realistic summaries
document.getElementById("ai-summary").textContent = "일기체 · 한국어 · 23:50";
document.getElementById("storage-summary").textContent = "로컬 · GitHub";
document.getElementById("data-summary").textContent = "ActivityWatch · 제외 앱 2개";
"""


def main():
    with sync_playwright() as p:
        browser = p.chromium.launch()
        ctx = browser.new_context(
            viewport={"width": 420, "height": 720},
            device_scale_factor=2,
        )
        page = ctx.new_page()
        page.add_init_script(TAURI_SHIM)
        page.goto(f"file://{HTML_PATH}")
        page.wait_for_timeout(500)
        page.evaluate(SHOW_SETTINGS)
        page.wait_for_timeout(300)

        out = OUT_DIR / "screenshot-settings.png"
        page.screenshot(path=str(out), type="png")
        print(f"✅ Settings screenshot saved: {out}")

        browser.close()


if __name__ == "__main__":
    main()

File: docs/screenshots/take-screenshot.mjs
import { chromium } from "playwright";
import { fileURLToPath } from "url";
import path from "path";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const htmlPath = path.resolve(__dirname, "../../apps/agent/src/index.html");
const outputPath = path.resolve(__dirname, "screenshot-settings.png");

(async () => {
  const browser = await chromium.launch();
  const context = await browser.newContext({
    viewport: { width: 420, height: 720 },
    deviceScaleFactor: 2,
  });
  const page = await context.newPage();

  // Suppress Tauri API errors
  await page.addInitScript(() => {
    window.__TAURI__ = {
      core: { invoke: async () => ({}) },
      event: { listen: async () => () => {} },
      shell: { open: async () => {} },
    };
  });

  await page.goto(`file://${htmlPath}`);
  await page.waitForTimeout(300);

  // Manipulate DOM to show settings tab
  await page.evaluate(() => {
    // Hide onboarding
    const onb = document.getElementById("onboarding-screen");
    if (onb) onb.classList.add("hidden");

    // Show tab bar
    const tabBar = document.querySelector(".tab-bar");
    if (tabBar) tabBar.style.display = "";

    // Activate settings tab
    document.querySelectorAll(".tab-content").forEach((c) => c.classList.remove("active"));
    const settingsTab = document.getElementById("tab-settings");
    if (settingsTab) settingsTab.classList.add("active");

    // Show settings main panel
    document.querySelectorAll(".view-panel").forEach((p) => p.classList.remove("active"));
    const settingsMain = document.getElementById("settings-main");
    if (settingsMain) settingsMain.classList.add("active");

    // Update tab bar icons
    document.querySelectorAll(".tab-bar-item").forEach((b) => b.classList.remove("active"));
    const settingsBtn = document.querySelector('.tab-bar-item[data-nav="settings"]');
    if (settingsBtn) settingsBtn.classList.add("active");

    // Set realistic summary texts
    const aiSummary = document.getElementById("ai-summary");
    if (aiSummary) aiSummary.textContent = "일기체 · 한국어 · 23:50";

    const storageSummary = document.getElementById("storage-summary");
    if (storageSummary) storageSummary.textContent = "로컬 · GitHub";

    const dataSummary = document.getElementById("data-summary");
    if (dataSummary) dataSummary.textContent = "ActivityWatch · 제외 앱 2개";
  });

  await page.waitForTimeout(200);
  await page.screenshot({ path: outputPath, type: "png" });
  console.log("Screenshot saved:", outputPath);

  await browser.close();
})();
